### PR TITLE
Koha: fixes failing search case where %3A, i.e. : fails

### PR DESCRIPTION
**Pull request by mekarpeles**
Addresses a case @JeffKaplan reported from project Koha about certain MARC records which contain ":" colons causing issues.

### CODE REVIEW

File: openlibrary/plugins/worksearch/code.py [hunk 1]
```diff
@@ -114,6 +114,21 @@
 plurals = dict((f + 's', f) for f in ('publisher', 'author'))
 
 
+def depunctuate(s):
```
Comment (cclauss) [inline openlibrary/plugins/worksearch/code.py:117]: [punctuation is `!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~`](https://docs.python.org/3.9/library/string.html#string.punctuation) in Python so perhaps a more self-documenting function name...
```suggestion
def colonoscopy(s):
```

File: openlibrary/plugins/worksearch/code.py [hunk 2]
```diff
@@ -114,6 +114,21 @@
...
+    """A developer from project Koha noticed Open Library search failing
+    for queries of the form:
+    openlibrary.org/search?author=Potter+Will&title=Green+is+the+new+red+%3A
+
+    This method attempts to sanitize fields of solr queries (for now,
+    just my removing `:`'s and stripping trailing whitespace.
```
Comment (cclauss) [inline openlibrary/plugins/worksearch/code.py:123]:
```suggestion
    just by removing `:`'s and stripping leading and trailing whitespace.
```

File: openlibrary/plugins/worksearch/code.py [hunk 3]
```diff
@@ -114,6 +114,21 @@
...
+
+    :param str s: a solr field value (e.g. a title)
+    :rtype: string
+    :return: the sanitized string
+    """
```
Comment (cclauss) [inline openlibrary/plugins/worksearch/code.py:128]:
```suggestion
    >>> depunctuate('  ! " #$ :A:B:C: %& ` ?  ')
    '! " #$ ABC %& ` ?'
    """
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: openlibrary/plugins/worksearch/code.py [hunk 4]
```diff
@@ -278,7 +293,7 @@ def build_q_list(param):
                     q_list.append("(author_name:(%(name)s) OR author_alternative_name:(%(name)s))" % {'name': v})
 
         check_params = ['title', 'publisher', 'oclc', 'lccn', 'contribtor', 'subject', 'place', 'person', 'time']
-        q_list += ['%s:(%s)' % (k, param[k]) for k in check_params if k in param]
+        q_list += ['%s:(%s)' % (k, depunctuate(param[k])) for k in check_params if k in param]
```
Comment (cclauss) [inline openlibrary/plugins/worksearch/code.py:296]:
```suggestion
        q_list += [
            '%s:(%s)' % (k, depunctuate(param[k])) for k in check_params if k in param
        ]
```
Use psf/black for format long line to placate Travis CI.

**Comment by cdrini** [comment]
Closing in favour of #4009

**Pull request closed without merging**